### PR TITLE
Updated to dotnet 8.0, fixed linter warnings and refactored tests.

### PR DESCRIPTION
**Pull request by jsaarimaa**
Updated to dotnet 8.0
Fixed the linter issues that emerged after the update.
Refactored some unit tests to use translation fixture instead of reaading the translation json

### CODE REVIEW

File: UnitTests/ChartTypeSelectionTests/GroupVerticalBarChartTests.cs [hunk 1]
```diff
@@ -1,4 +1,6 @@
-﻿using NUnit.Framework;
+﻿using Microsoft.AspNetCore.Hosting.StaticWebAssets;
```
Comment (sakari-malkki) [inline UnitTests/ChartTypeSelectionTests/GroupVerticalBarChartTests.cs:1]: This appears to be here by accident, doesn't seem it's needed

Comment (sakari-malkki) [reply UnitTests/ChartTypeSelectionTests/GroupVerticalBarChartTests.cs:1]: System.Linq seems to be unused too a few lines below

**Review by sakari-malkki — APPROVED** [review]
Left a comment or two about unnecessary imports in one file. Apart from that everything seems to be in order

File: UnitTests/ChartTypeSelectionTests/GroupVerticalBarChartTests.cs (hunk 1 above)
Comment (jsaarimaa) [reply UnitTests/ChartTypeSelectionTests/GroupVerticalBarChartTests.cs]: Removed the unused statements

**Review by sakari-malkki — APPROVED** [review]

**Pull request merged**